### DISCUSSION TITLE
Unable to Download Videos from Panda Video Platform Using Video DownloadHelper

### DISCUSSION THREAD

**Discussion by damujen** [aclap-dev/video-downloadhelper#1467] [Help and Q&A]
### Your question

Hi everyone,

I've been experiencing an issue with Video DownloadHelper where I'm unable to download videos hosted by Panda Video. Whenever I attempt to download, it doesn't seem to capture or process the video link properly. It gives me severall instances of a 6 seconds video. I've tried a few troubleshooting steps, like restarting the browser and updating the add-on, but nothing seems to work. I've also tried a BETA version on Firefox.

Has anyone else experienced this problem? Are there any known solutions or workarounds? Additionally, is the development team aware of this issue, and are there plans to support Panda Video in future updates?

Thank you for your assistance.

### Addon & CoApp version and details

version: 9.1.1.0
target: google
channel: stable
lang: pt-PT
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"08cac07f20e446bc"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "concurrent_downloads_max": 100,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 8,
    "successfull_dl": 6,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_2838396654578935"
                    },
                    {
                        "download_result": {
                            "filedir": {
                                "__serializer_tag": "primitive",

**Comment by Mattmjs** [comment]
Can you follow these steps carefully

1. Click the 3 dots next to download and look down to details.
2. Click this, a page open with more information about the media.
3. **Ctrl + A** to select all on page.
4. **Ctrl + C** to copy information to the clipboard.
5. **Ctrl + V** to paste the information here in your reply.

**Reply by damujen** [reply]
<html>
<body>

Tab | 1629378973 (incognito: false)
-- | --
Aula 01 - Vida intelectual \| Parte 1 - Comunidade Filosofia do Zero
https://comunidade.filosofiadozero.com.br/area/produto/item/2264254 with 12 headers
https://b-vz-d031816a-48f.tv.pandavideo.com.br/1fb223a6-4118-4872-9bed-4f743e0ea013/thumbnail.jpg
false
https://b-vz-d031816a-48f.tv.pandavideo.com.br/1fb223a6-4118-4872-9bed-4f743e0ea013/1280x720/video.m3u8?token=1AfPHG5NyuRKatcnT0gziYvn1kZmMkx904Px0VVtE7TSk5DQbr9wOJutCmYwfC7L
https://b-vz-ded14ebd-85a.tv.pandavideo.com.br/cd676325-3f2d-4ad5-b95c-c2a683d7b0cc/1280x720/video.m3u8
none
Mp4 b:Hls p:hls 6.066667s
unknown at 2800000, fps: ?, q:720, {"height":720,"width":1280}
https://b-vz-d031816a-48f.tv.pandavideo.com.br/1fb223a6-4118-4872-9bed-4f743e0ea013/1280x720/video.m3u8?token=1AfPHG5NyuRKatcnT0gziYvn1kZmMkx904Px0VVtE7TSk5DQbr9wOJutCmYwfC7L
https://b-vz-ded14ebd-85a.tv.pandavideo.com.br/cd676325-3f2d-4ad5-b95c-c2a683d7b0cc/842x480/video.m3u8
none
Mp4 b:Hls p:hls 6.066667s
unknown at 1400000, fps: ?, q:480, {"height":480,"width":842}


</body>
</html>

**Reply by Mattmjs** [reply]
I see what you're saying, I could only detect a 6 second video. The video says it cannot be shared. I think the next step is to record
the network activity and attach it here in your reply. Visit your video again and wait until the page has stopped loading.
Then open the developer tools using Ctrl+shift+I or F12 and select network if it isn't already. Then reload the page and once it 
stops click the button shown in the image to save it. It needs to be attach here as a .zip file.

![chrome dev tools](https://github.com/user-attachments/assets/0a57d5b5-b8aa-4208-b7ea-eeea219661c2)

**Reply by damujen** [reply]
Thank you, I'm learning a lot. 
I've got a .har file and zipped it to upload here, but it turned out to be bigger than 25mb which is the file size limit.
So'I've uploaded it to m y google drive, I hope it helps.
.zip: https://drive.google.com/file/d/1ZnCMgyq_0Zm6y8Y98L_n7nyTp36J1XRY/view?usp=sharing
.har: https://drive.google.com/file/d/1DjD0PFo8ZJAT3qZ0HFJDdDnVj2E4ko6b/view?usp=sharing

**Reply by Mattmjs** [reply]
Thanks, this helps for more investigating but I'm not sure there could be a solution. I found an embed link which opens your video.
It detects the full video at 19:48 minutes but when launching a download it's still the 6 second cannot be shared video.
Inside one of the streaming files is this:   **#EXTINF:0,duration:1189|drm:false|block_download:true|percent_ts:0**

Some mechanism implemented to prevent download of the media. The .ts chunks are not protected , you could download these
and concat them in a single file with ffmpeg but that would be an arduous task with there being 297 chunks.
Here is the embed link for others to check out : 
https://player-vz-d031816a-48f.tv.pandavideo.com.br/embed/?color=8a1a12&v=5b858a22-915a-4493-9fc2-dfb74da90470

@paulrouget

**Reply by Mattmjs** [reply]
On their website is another example: https://www.pandavideo.com.br/

**Reply by damujen** [reply]
Thank you for trying! I have a lot to learn before I can download all those tiny videos and stitch them together with ffmpeg, but 'm glad to know I've found a real issue. Maybe one day the program will be able to handle pandavideo again. Thanks

**Reply by Mattmjs** [reply]
You can scrub that idea as I think there is a solution that works. I went back to the old companion app version 1.6.3 and extension
version 8.0.0.6 on a Linux virtual machine. Then I had a look at the details page of a download that worked. Image below

![Screenshot from 2024-11-25 01-28-12](https://github.com/user-attachments/assets/51d8cf4a-7eb1-466c-a6a4-f471b5c6df2a)

Then back on windows with the versions 9.1.0.44 (stable) + Companion App installed (v2.0.19) I looked at what the details show in
the image below. You can see that the links have a token attached to the urls in the current software but on the old software
it omitted the tokens. So copying one of the streaming links without the token and using it in ffmpeg the complete file is
downloaded. Therefore this can be fixed in VDH. You can try yourself by opening one of those streaming links in a new tab
without the token and VDH should detect it as well as getting the whole file.

![Screenshot 2024-11-25 175100](https://github.com/user-attachments/assets/deae0f26-5253-4b64-9c3c-fde7b7197a1f)

**Reply by nerokingdom** [reply]
Thanks, that worked. I download the video using yt-dlp after taking the token part. 

And for a mind-blowing coincidence... i was triyng to download a video from the same course from OP. Filosofia na veia.

**Comment by Mattmjs** [comment]
There was an example of this on the old google groups forum. And it worked then on the old companion app.
I tried in linux mint using the old extension + companion app and it seems to work.

**Reply by rafo-br** [reply]
I noticed that, for the same video, VDH cannot detect it on linux but it does detect it on windows. I use firefox on both OS. Does anyone have a clue why that happens?
